src/controllers/userController.js

```diff
@@ -100,4 +100,54 @@ const logOut = (req, res) => {
   }
 };
 
-module.exports = { signUp, logIn, logOut };
+/******************************************************
+ * @UPDATEUSER
+ * @route /api/v1/updateuser
+ * @method PUT
+ * @description singUp function for creating new user
+ * @body name, email, password, confirmPassword
+ * @returns User Object
+ ******************************************************/
+
+const updateUser = async (req, res) => {
+  try {
+    const { userId } = req.user; //retrieve user.id from jwtverify()
+    const user = await userModel.findByIdAndUpdate(userId, req.body);
```

Comment:
If we pass user password in body and just update it in the database then it gets updated and becomes the new password, when it should get hashed before being sent to the database.